### PR TITLE
Docs: Testing tasks as well as flows

### PR DESCRIPTION
**Pull request by dannysepler**
## Summary
Following up from [this slack thread](https://prefect-community.slack.com/archives/CL09KU1K7/p1650416317687919), this should explain how to unit test tasks, in addition to flows.

Open question: should I explain more why someone may want to use `.fn()`, or is it pretty clear?

## Changes
Adds a bit more to the Testing section.

## Importance
To help folks know the options for their unit tests. It took me a while to realize that you didn't need a `flow` for unit tests. They were adding some runtime to our test suite, when the functions themselves were pretty simple!

## Checklist

This PR:

- [ ] adds new tests (if appropriate)
- [ ] adds a change file in the `changes/` directory (if appropriate)
- [ ] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

### CODE REVIEW

**Review by desertaxle — CHANGES_REQUESTED** [review]
I have one nit, but overall this looks great!

File: docs/tutorials/testing.md [hunk 1]
```diff
@@ -38,13 +37,42 @@ For more extensive testing, you can leverage `prefect_test_harness` as a fixture
...
+@task
+def my_favorite_task():
+    return 42
+
+def test_my_favorite_task():
+    @flow
+    def my_flow():
+        return my_favorite_task()
+    
+    state = my_flow().result()
```
Comment (desertaxle) [inline docs/tutorials/testing.md:67]: Nit: Naming this variable `task_state` disambiguates with state `my_flow().result()` is returning.
```suggestion
    task_state = my_flow().result()
```

**Review by desertaxle — APPROVED** [review]
LGTM!

**Review by tpdorsey — APPROVED** [review]
Looks good. Thank you!

**Pull request merged**